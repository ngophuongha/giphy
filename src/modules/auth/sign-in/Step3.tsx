import { Form, InputGroup } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPhone } from "@fortawesome/free-solid-svg-icons";
import { useTranslation } from "react-i18next";

import { useAppDispatch } from "../../../store/hooks";
import { StyledButton as Button } from "../../../components/button";
import { routesConstants } from "../../../pages/routes.constant";
import { SignInStepProps } from "./SignIn.form.type";

import { StyledSignInStep } from "./SignIn.form.styled";
import { setAuth } from "../../../store/auth/auth";
export const Step3 = ({
  onSuccess,
}: //   onFailed,
SignInStepProps): JSX.Element => {
  const { t } = useTranslation("common");
  const dispatch = useAppDispatch();

  const setAuthInfo = async () => {
    await dispatch(
      setAuth({
        name: "Ha Ngo Phuong",
        avatar: "",
      })
    );
    onSuccess();
  };

  return (
    <StyledSignInStep
      initial={{ x: "-500px" }}
      animate={{ translateX: "500px" }}
      transition={{ stiffness: 150 }}
    >
      <h6 className="d-flex justify-content-center fw-normal mb-3 mt-2">
        {t("auth.enterCode")}
      </h6>
      <Form>
        <InputGroup className="my-3">
          <InputGroup.Text>
            <FontAwesomeIcon icon={faPhone} />
          </InputGroup.Text>
          <Form.Control
            name="password"
            placeholder={t("auth.placeholder.enterCode")}
            type="password"
          />
        </InputGroup>
        <div className="d-grid">
          <Button
            variant="primary"
            btnTitle={t("auth.signInBtn")}
            type="submit"
            onClick={setAuthInfo}
          />
        </div>
        <div className="d-flex justify-content-center mt-2">
          <Form.Text>
            {t("auth.notRegistered")}{" "}
            <a href={routesConstants.auth.register}>
              {t("auth.createNewAccount")}
            </a>
          </Form.Text>
        </div>
      </Form>
    </StyledSignInStep>
  );
};
