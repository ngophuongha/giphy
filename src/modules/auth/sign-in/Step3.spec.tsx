import { render } from "../../../../test";
import { Step3 } from "./Step3";

describe("Sign In: Step 3", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Step3 onSuccess={jest.fn()} onFailed={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
