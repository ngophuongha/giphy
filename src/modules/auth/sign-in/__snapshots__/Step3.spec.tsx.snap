// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In: Step 3 should match snapshot 1`] = `
.c1 {
  padding: 0.25rem !important;
  border-radius: 0.25rem;
  border: 1px solid #2b5dab;
  color: #fff;
  background-color: #2b5dab;
}

.c1:hover {
  background-color: #25539c;
  color: #fff;
}

.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
}

<div>
  <div
    class="c0"
    style="transform: translateX(-500px) translateZ(0);"
  >
    <h6
      class="d-flex justify-content-center fw-normal mb-3 mt-2"
    >
      auth.enterCode
    </h6>
    <form
      class=""
    >
      <div
        class="my-3 input-group"
      >
        <span
          class="input-group-text"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-phone fa-w-16 "
            data-icon="phone"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          class="form-control"
          name="password"
          placeholder="auth.placeholder.enterCode"
          type="password"
        />
      </div>
      <div
        class="d-grid"
      >
        <button
          class="c1"
          type="submit"
          variant="primary"
        >
          auth.signInBtn
        </button>
      </div>
      <div
        class="d-flex justify-content-center mt-2"
      >
        <small
          class="form-text"
        >
          auth.notRegistered
           
          <a
            href="/register"
          >
            auth.createNewAccount
          </a>
        </small>
      </div>
    </form>
  </div>
</div>
`;
